### PR TITLE
Create esx_hud

### PR DESCRIPTION
**Pull request by Rangerhitam12**
[Uploading esx_hud.zip…]()
added the function of the stress icon

### CODE REVIEW

File: shared/config.lua [hunk 1]
```diff
@@ -11,6 +11,7 @@ Config = {
             armorBar = "blue",
             drinkBar = "lightblue",
             foodBar = "yellow",
```
Comment (Gellipapa) [inline shared/config.lua:13]: Where drunk bar?

File: web/DefaultDatas.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ const progressColors = {
     foodBar: "yellow",
     oxygenBar: "green",
     staminaBar: "purple",
+    stressBar: "orange"
```
Comment (Gellipapa) [inline web/DefaultDatas.js:32]: where drunk bar?

File: web/DefaultDatas.js [hunk 3]
```diff
@@ -64,6 +65,7 @@ const progressLevels = {
     foodBar: 80,
     oxygenBar: 100,
     staminaBar: 100,
+    stressBar: 50, // Tambahkan nilai sesuai kebutuhan
```
Comment (Gellipapa) [inline web/DefaultDatas.js:68]: remove comment, where drunkBar?

File: web/DefaultDatas.js [hunk 4]
```diff
@@ -128,6 +130,18 @@ const progressDefaultCircles = [
         color: "green",
         icon: StaminaIcon,
     },
+    {
+        name: "stressBar",
+        progressLevel: 100,
+        color: "orange",
+        icon: StressIcon, // Pastikan Anda memiliki ikon untuk stress
```
Comment (Gellipapa) [inline web/DefaultDatas.js:137]: remove comment

File: web/DefaultDatas.js [hunk 5]
```diff
@@ -128,6 +130,18 @@ const progressDefaultCircles = [
...
+    },
+    {
+        name: "drunkBar",
+        progressLevel: 100,
+        color: "purple",
+        icon: DrunkIcon, // Pastikan Anda memiliki ikon untuk drunk
```
Comment (Gellipapa) [inline web/DefaultDatas.js:143]: remove comment

File: web/src/assets/translate.json [hunk 6]
```diff
@@ -150,7 +153,9 @@
       "foodbar": "Étel státusz",
       "oxygenbar": "Oxigén státusz",
       "staminabar": "Stamina státusz",
-      "circlewidth": "Státusz kör vastagsága"
+      "stressbar": "Stress bar",
+      "drunkbar": "Drunk bar",
+      "circlewidth": "Állapot körszélesség"
```
Comment (Gellipapa) [inline web/src/assets/translate.json:158]: use old circleWidth label : "Státusz kör vastagsága"

**Review by Gellipapa — CHANGES_REQUESTED** [review]
I think that these are not normal statuses so you should write the logic to disable this in config.lua, I mean the stress and drunk hud part.

File: fxmanifest.lua [hunk 7]
```diff
@@ -3,7 +3,7 @@ game 'gta5'
 lua54 'yes'
 
 description 'The default HUD resource for ESX-Legacy.'
-version '1.7.0'
+version '1.8.0'
```
Comment (Gellipapa) [inline fxmanifest.lua:6]: We have automatic github actions, and change the version number if we released new version, please revert to old. (1.7.0)

**Review by Rangerhitam12 — COMMENTED** [review]
if not Config.Disable.Status then
    local values = {}
    local playerId = PlayerId()

    AddEventHandler("esx_status:onTick", function(data)
        local hunger, thirst, stress
        for i = 1, #data do
            if data[i].name == "thirst" then
                thirst = math.floor(data[i].percent)
            end
            if data[i].name == "hunger" then
                hunger = math.floor(data[i].percent)
            end
            if data[i].name == "stress" then
                stress = math.floor(data[i].percent)
            end
        end

        local ped = PlayerPedId()

        values.healthBar = math.floor((GetEntityHealth(ped) - 100) / (GetEntityMaxHealth(ped) - 100) * 100)
        values.armorBar = GetPedArmour(ped)
        values.drinkBar = thirst
        values.foodBar = hunger
        if not Config.Disable.Stress then
            values.stressBar = stress
        end
    end)

    function HUD:StatusThread()
        values = {}
        CreateThread(function()
            while ESX.PlayerLoaded do
                local oxygen, stamina, drunk
                oxygen = math.floor(GetPlayerUnderwaterTimeRemaining(playerId) * 10)
                stamina = math.floor(100 - GetPlayerSprintStaminaRemaining(playerId))
                if stamina == 0 then
                    stamina = 1
                end
                if stamina == 100 then
                    stamina = 0
                end

                -- Add drunk status calculation here
                drunk = math.floor(ESX.Game.GetPlayerDrunkLevel(playerId) or 0)  -- Assuming there is a method to get the player's drunk level

                values.oxygenBar = IsPedSwimmingUnderWater(PlayerPedId()) and oxygen or 0
                values.staminaBar = stamina
                if not Config.Disable.Drunk then
                    values.drunkBar = drunk  -- Add drunk level to the values table
                end

                SendNUIMessage({ type = "STATUS_HUD", value = values })
                Wait(200)
            end
        end)
    end
end

File: web/src/assets/Contexts/HudStorage.jsx [hunk 8]
```diff
@@ -42,6 +42,18 @@ const initialState = {
...
+            name: "stressBar",
+            progressLevel: 100,
+            color: "orange",
+            icon: StressIcon // Pastikan Anda memiliki ikon untuk stress
+        },
+        {
+            name: "drunkBar",
+            progressLevel: 100,
+            color: "purple",
+            icon: DrunkIcon // Pastikan Anda memiliki ikon untuk drunk
```
[the remark this answers is not in the document]
Comment (Gellipapa) [reply web/src/assets/Contexts/HudStorage.jsx:56]: remove comments

File: web/src/assets/translate.json (hunk 6 above)
Comment (Gellipapa) [reply web/src/assets/translate.json:159]: use old circleWidth label : "Státusz kör vastagsága" please use old label "Státusz kör vastagsága"

**Review by Gellipapa — CHANGES_REQUESTED** [review]
Please do this changes

**Comment by Gellipapa** [comment]
@Rangerhitam12 This doesn't solve the problem of not displaying the HUD in the UI interface, this has to be handled on the UI side as well so that the two statuses can be set based on the config.

**Comment by Gellipapa** [comment]
@Rangerhitam12 Please remove all comments in codebase like "add this line", etc.

**Pull request closed without merging**